src/Asana/AsanaApiClient.php

```diff
@@ -193,4 +203,107 @@ public function createCard(string $projectId, array $values): void
+                'custom_fields'.'['.$this->options['yesplan_status'].']' => $values['status'],
+                'custom_fields'.'['.$this->options['yesplan_profile'].']' => $values['profile'],
+                'projects' => $projectId,
+            ],
+        ];
+
+            $response = $this->post($url, $options);
+            if (!(Response::HTTP_CREATED === $response->getStatusCode())) {
+                $this->mailer->sendEmail('Error creating card', 'Error '.$response->getStatusCode().'URL: '.$url.'projectID: '.$projectId);
+                $this->error('Card not created {status_code}, response {response}', ['status_code' => $response->getStatusCode(), 'response' => $response]);
+            } else {
+                $this->debug('Card created yesplan_id: ', ['yesplan_id' => $this->options['yesplan_id']]);
+            }
+        }
+
+        //create red cards in calendar on presale date
+        if (!empty($values['presaleDate'])) {
+            $options = [
+            'body' => [
+                'name' => $values['title'],
+                'due_on' => $presaleDate,
+                'custom_fields'.'['.$this->options['asana_calendar_colorfield'].']' => $this->options['asana_calendar_colorfield_red'],
+                'custom_fields'.'['.$this->options['yesplan_id'].']' => $values['id'],
+                'custom_fields'.'['.$this->options['yesplan_eventDate'].']' => $eventDate,
+                'custom_fields'.'['.$this->options['yesplan_location'].']' => $values['location'],
+                'custom_fields'.'['.$this->options['yesplan_genre'].']' => $values['genre'],
+                'custom_fields'.'['.$this->options['yesplan_marketingBudget'].']' => $values['marketingBudget'],
+                'custom_fields'.'['.$this->options['yesplan_publicationDate'].']' => $publicationDate,
+                'custom_fields'.'['.$this->options['yesplan_presaleDate'].']' => $presaleDate,
+                'custom_fields'.'['.$this->options['yesplan_insaleDate'].']' => $insaleDate,
+                'custom_fields'.'['.$this->options['yesplan_percent'].']' => $values['percent'],
+                'custom_fields'.'['.$this->options['yesplan_status'].']' => $values['status'],
+                'custom_fields'.'['.$this->options['yesplan_profile'].']' => $values['profile'],
+                'projects' => $projectId,
+            ],
+        ];
+
+            $response = $this->post($url, $options);
+
+            if (!(Response::HTTP_CREATED === $response->getStatusCode())) {
+                $this->mailer->sendEmail('Error creating card', 'Error '.$response->getStatusCode().'URL: '.$url.'projectID: '.$projectId);
+                $this->error('Card not created {status_code}, response {response}', ['status_code' => $response->getStatusCode(), 'response' => $response]);
+            } else {
+                $this->debug('Card created yesplan_id: ', ['yesplan_id' => $this->options['yesplan_id']]);
+            }
+        }
```

Comment:
You have (almost) the same code three times here. Move it onto a function or use a loop to reuse the code for generating colored stuff.